components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java

```diff
@@ -7658,42 +7657,44 @@ public int addComment(APIIdentifier identifier, String commentText, String user)
     /**
      * Adds a comment for an API
      *
-     * @param identifier API Identifier
+     * @param apiTypeWrapper API type wrapper
      * @param comment    Commented Text
      * @param user       User who did the comment
      * @return Comment ID
      */
-    public String addComment(APIIdentifier identifier, Comment comment, String user) throws APIManagementException {
+    public String addComment(ApiTypeWrapper apiTypeWrapper, Comment comment, String user) throws APIManagementException {
         Connection connection = null;
-        ResultSet resultSet = null;
         ResultSet insertSet = null;
-        PreparedStatement getPrepStmt = null;
         PreparedStatement insertPrepStmt = null;
         String commentId = null;
-        int apiId = -1;
+        int id = -1;
+        Boolean isProduct = apiTypeWrapper.isAPIProduct();
```

Comment:
Lets make this the primitive _boolean_ type instead of _Boolean_

Reply:
Fixed